### ISSUE TITLE
Is this package still needed on Laravel 10.x

### ISSUE THREAD

**Issue by flavio-alves-roboyo** [fntneves/laravel-transactional-events#53]
I am using this amazing package for a long time, but I don't know if it is still needed on Laravel 10.x.

The laravel docs says:

> Setting the after_commit configuration option to true will also cause any queued event listeners, mailables, notifications, and broadcast events to be dispatched after all open database transactions have been committed.

@fntneves could you please help with this question?

**Comment by leeovery** [comment]
I don't believe this is needed anymore. 

We can now add interfaces to events, listeners and observers, as well as mark the queue connection to always dispatch after commit. Can also tell jobs and events being dispatched to do so after commit. 

With the right setup and tweaks to your stubs you can safely remove this great package now. 

Been using it since the beginning. Served a purpose. Now it can rest ❤️

**Comment by fntneves** [comment]
I agree with the @leeovery's comment. Laravel evolved to support the behaviour this package offered.

Now, I believe this package is not needed anymore. It was a great journey and it helped many folks.

Thank you for your support ❤

**Comment by mfn** [comment]
I'm also biding farewell to this great package 🫡 

It helped tremendously at time when Laravel wasn't on that level yet, but it now has caught up and at least the aspects of this package I used could be fully replaced:
- enable `after_commit=true` on the queue (e.g. redis in my case)
- use `->beforeCommit()` on jobs who should immediately dispatch
- use `ShouldDispatchAfterCommit` on events which should respect transactions too

Thanks for everything 🙇🏼

**Comment by fsmeier** [comment]
Doesnt it make sense to add a hint to the package readme then?

**Comment by bytestream** [comment]
Isn't the whole point of this package following the recent additions to Laravel that:
> Yet, it lacks transaction-aware behavior support for Eloquent events.

Therefore the package still has a purpose. I'm happy to help maintain it if no-one else is.

**Comment by mfn** [comment]
It's technically "possible" to do it with native Laravel, but it's not as integrated as the rest in Laravel: use observers or `Event::listen()` for Eloquent string events, and make the listener classes transaction-aware. This works for the `…ed` events (`created`, …).

Note: in my case I did not use Eloquent events, so this part was of no concern when I migrated away.
